### ISSUE TITLE
Retargeting asserts when attribute forwarded in NoPia scenario

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#65048]
The basic scenario here is when:

1. An initial compilation occurs that involves NoPia and a reference to `System.Runtime.InteropServices.dll` which houses `PrimaryInteropAssemblyAttribute`. 
2. A subsequent compilation using the reference generated from 1, against a new set of references but `PrimaryInteropAssemblyAttribute` was forwarded to a new assembly. 

A working repro can be found by doing the following: 

- `git remote add jaredpar https://github.com/jaredpar/roslyn.git`
- `git fetch jaredpar`
- `git checkout -B repro jaredpar/repro-65048` 
- Execute `DefaultInterfaceImplementationTests.NoPia_09` on `net472` 

The retargeting layer cannot find the appropriate constructor as the assembly is missing. This triggers an assert in `SourceAttributeData` as it's expecting either a symbol or an error 

```csharp
Debug.Assert(attributeConstructor is object || hasErrors);
```

The stack trace of the assert is as follows

```
    ThrowingTraceListener.Fail(String message, String detailMessage) line 26
    TraceInternal.Fail(String message)
    Debug.Assert(Boolean condition)
    SourceAttributeData.ctor(SyntaxReference applicationNode, NamedTypeSymbol attributeClass, MethodSymbol attributeConstructor, ImmutableArray`1 constructorArguments, ImmutableArray`1 constructorArgumentsSourceIndices, ImmutableArray`1 namedArguments, Boolean hasErrors, Boolean isConditionallyOmitted) line 44
    RetargetingAttributeData.ctor(SyntaxReference applicationNode, NamedTypeSymbol attributeClass, MethodSymbol attributeConstructor, ImmutableArray`1 constructorArguments, ImmutableArray`1 constructorArgumentsSourceIndices, ImmutableArray`1 namedArguments, Boolean hasErrors, Boolean isConditionallyOmitted) line 24
    RetargetingSymbolTranslator.RetargetAttributeData(CSharpAttributeData oldAttributeData) line 1216
    <>c.<RetargetAttributes>b__45_0(CSharpAttributeData a, RetargetingSymbolTranslator t) line 1167
    ImmutableArray.CreateRange[TSource,TArg,TResult](ImmutableArray`1 items, Func`3 selector, TArg arg)
    ImmutableArrayExtensions.SelectAsArray[TItem,TArg,TResult](ImmutableArray`1 items, Func`3 map, TArg arg) line 156
    RetargetingSymbolTranslator.RetargetAttributes(ImmutableArray`1 oldAttributes) line 1167
    <5 more frames...>
    CSharpCompilation.GetSourceDeclarationDiagnostics(SyntaxTree syntaxTree, Nullable`1 filterSpanWithinTree, Func`4 locationFilterOpt, CancellationToken cancellationToken) line 2959
    CSharpCompilation.GetDiagnosticsWithoutFiltering(CompilationStage stage, Boolean includeEarlierStages, BindingDiagnosticBag builder, CancellationToken cancellationToken) line 2729
    CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, DiagnosticBag diagnostics, CancellationToken cancellationToken) line 2655
    CSharpCompilation.GetDiagnostics(CompilationStage stage, Boolean includeEarlierStages, CancellationToken cancellationToken) line 2647
    CSharpCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean emitMetadataOnly, Boolean emitTestCoverageData, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) line 3201
    Compilation.Emit(Stream peStream, Stream metadataPEStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, RebuildData rebuildData, CompilationTestData testData, CancellationToken cancellationToken) line 2890
    CompilationExtensions.EmitToArray(Compilation compilation, EmitOptions options, CompilationTestData testData, DiagnosticDescription[] expectedWarnings, Stream pdbStream, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, IEnumerable`1 manifestResources, Stream metadataPEStream) line 74
    CompilationExtensions.EmitToPortableExecutableReference(Compilation comp, EmitOptions options, Boolean embedInteropTypes, ImmutableArray`1 aliases, DiagnosticDescription[] expectedWarnings) line 128
    CompilationExtensions.EmitToImageReference(Compilation comp, EmitOptions options, Boolean embedInteropTypes, ImmutableArray`1 aliases, DiagnosticDescription[] expectedWarnings) line 119
    DefaultInterfaceImplementationTests.NoPia_09() line 50583
```
